File: backend/app/schemas/v2_home.py
import uuid

from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict

from app.models.enums import WorkspaceColor, WorkspaceIcon


class _StrictModel(BaseModel):
    model_config = ConfigDict(extra="forbid")


class HomeTodayCounts(_StrictModel):
    tasks: int
    pending_items: int
    project_stages: int
    activities: int


class HomeWorkspace(_StrictModel):
    id: uuid.UUID
    name: str
    color: WorkspaceColor
    icon: WorkspaceIcon


class HomeUpcomingActivity(_StrictModel):
    id: uuid.UUID
    workspace: HomeWorkspace
    name: str
    starts_at: datetime
    ends_at: datetime


class HomeAttentionItem(_StrictModel):
    type: Literal["TASK", "PENDING_ITEM", "PROJECT_STAGE"]
    id: uuid.UUID
    workspace: HomeWorkspace
    name: str
    planned_date: date
    project_id: uuid.UUID | None = None


class HomeUpcomingDay(_StrictModel):
    date: date
    tasks: int
    pending_items: int
    project_stages: int
    activities: int


class HomeSummaryRead(_StrictModel):
    local_date: date
    today: HomeTodayCounts
    upcoming_activities: list[HomeUpcomingActivity]
    attention: list[HomeAttentionItem]
    upcoming_days: list[HomeUpcomingDay]

File: frontend/src/types/v2Home.ts
export type HomeWorkspace = { id: string; name: string; color: string; icon: string };

export type HomeTodayCounts = {
  tasks: number;
  pending_items: number;
  project_stages: number;
  activities: number;
};

export type HomeUpcomingActivity = {
  id: string;
  name: string;
  starts_at: string;
  ends_at: string;
  workspace: HomeWorkspace;
};

export type HomeAttentionItem = {
  type: "TASK" | "PENDING_ITEM" | "PROJECT_STAGE";
  id: string;
  name: string;
  planned_date: string;
  workspace: HomeWorkspace;
  project_id: string | null;
};

export type HomeUpcomingDay = HomeTodayCounts & { date: string };

export type V2HomeSummary = {
  local_date: string;
  today: HomeTodayCounts;
  upcoming_activities: HomeUpcomingActivity[];
  attention: HomeAttentionItem[];
  upcoming_days: HomeUpcomingDay[];
};
